operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java

```diff
@@ -71,15 +72,22 @@ public Controller(Reconciler<P> reconciler,
     eventSourceManager = new EventSourceManager<>(this);
 
     final var hasDeleterHolder = new boolean[] {false};
-    dependents = configuration.getDependentResources().stream()
-        .map(drs -> createAndConfigureFrom(drs, kubernetesClient))
-        .peek(d -> {
-          // check if any dependent implements Deleter to record that fact
-          if (!hasDeleterHolder[0] && d instanceof Deleter) {
-            hasDeleterHolder[0] = true;
-          }
-        })
-        .collect(Collectors.toList());
+    final var specs = configuration.getDependentResources();
+    final var size = specs.size();
+    if (size == 0) {
```

Comment:
Please extract this part into a separate method, I mean the code block to initiate the dependents.

Reply:
Moving it to a different methods would prevent making the `dependents` field `final` and to be honest, I'm not sure what purpose introducing another method would serve.

Reply:
Readability, I would also reintroduc DependentResourceManager, since there is now more considerable logic.